### ISSUE TITLE
Metadata is missing from reports

### ISSUE THREAD

**Issue by GriffinSauce** [xojs/xo#511]
I'm writing a formatter based on the ESLint spec only to discover that the metadata is not returned from XO.
That blocks quite a lot of value.

In [`mergeReports`](https://github.com/xojs/xo/blob/master/index.js#L21) the metadata should be merged as well and then passed as the second parameter in [`log`](https://github.com/xojs/xo/blob/master/cli-main.js#L144).

Background:
https://eslint.org/docs/developer-guide/working-with-custom-formatters#the-data-argument

This currently already breaks the `json-with-metadata` reporter.

**Comment by sindresorhus** [comment]
I agree. It's not something I plan to work on, but a pull request is welcome.

**Comment by sindresorhus** [comment]
This was fixed back in July 2022 via #674. The `rulesMeta` is now extracted using `eslint.getRulesMetaForResults()` and passed to formatters as part of the data argument.
